docs/contracts/storage-layout-specifier.rst

```diff
@@ -0,0 +1,61 @@
+.. index:: ! storage layout specifier, ! layout at
+
+.. _storage-layout-specifier:
+
+************************
+Storage Layout Specifier
+************************
+
+Contracts can define an arbitrary base slot for its own storage.
+The contract's state variables will be stored from the specified slot
+instead of the default slot zero.
+This can be done by using ``layout at <base-slot-expression>`` in the header of
+a contract definition, either after or before the inheritance specifier.
+It can be given at most once.
+The ``base-slot-expression`` must be an expression consisting of number literals
+that can be evaluated at compile time and yield in a rational value in the range of ``uint256``.
+As a general rule, any expression that can be used to specify the size of an array
+may also be used to indicate a slot.
+The layout of a contract without an explicit specification is identical to one with ``layout at 0``.
+The storage layout cannot be specified for abstract contracts, interfaces and libraries.
```

Comment:
We should also say that transient storage layout is not affected by the specifier (may not be obvious - someone might assume it shifts both).